File: wordBreak.cpp
class Solution {
public:
    bool wordBreak(string s, vector<string>& wordDict) {
        vector<int> result(s.size()+1,0);
        for(int i=0;i<s.size();i++){
            for(int j=i+1;j<s.size()+1;j++){
                if(i>0&&result[i]!=1){
                    break;
                }
                else{
                    auto iter = find(wordDict.begin(),wordDict.end(),s.substr(i,j-i));
                    if(iter!=wordDict.end()){
                        result[j] = 1;
                    }
                }
            }
        }
        if(result[s.size()]==1){
            return true;
        }
        else{
            return false;
        }
    }
};

File: wordBreak.py
class Solution:
    # 动态规划，使用一个数组来表示s[i:j]是否在wordDict上。
    def wordBreak(self, s: str, wordDict):
        dp = [0 for i in range(len(s) + 1)] 
        for i in range(len(s)):
            for j in range(i + 1, len(s) + 1):
                if i > 0 and dp[i] != 1:
                    break
                if s[i:j] in wordDict:
                    dp[j] = 1
        if dp[len(s)]:
            return True
        return False 
